sormas-e2e-tests/src/test/java/org/sormas/e2etests/steps/web/application/cases/CreateNewCaseSteps.java

```diff
@@ -18,29 +18,265 @@
+          webDriverHelpers.clickOnWebElementBySelector(BACK_TO_THE_CASES_LIST_BUTTON);
+        });
+
+    When(
+        "I filter by ([^\"]*) user condition",
+        (String userCondition) -> {
+          if (userCondition.equals("unspecified")) {
+            webDriverHelpers.selectFromCombobox(CASE_PRESENT_CONDITION_COMBOBOX, "");
+          } else {
+            webDriverHelpers.selectFromCombobox(CASE_PRESENT_CONDITION_COMBOBOX, userCondition);
+          }
+          webDriverHelpers.clickOnWebElementBySelector(CASE_APPLY_FILTERS_BUTTON);
+          TimeUnit.SECONDS.sleep(2); // Wait for filter
+          webDriverHelpers.waitForPageLoadingSpinnerToDisappear(40);
+          webDriverHelpers.waitForPageLoaded();
+        });
+
+    When(
+        "I check if created user is on filtered list with ([^\"]*) status as a ([^\"]*) record in the table",
+        (String status, String record) -> {
+          List<Map<String, String>> tableRowsData = getTableRowsData();
+          Map<String, String> detailedCasesTableRow;
+          int rec = Integer.parseInt(record);
+          int selectedRecord = 1;
+          Integer rec1 = rec - selectedRecord;
+          detailedCasesTableRow = tableRowsData.get(rec1);
+          softly.assertEquals(
+              status,
+              detailedCasesTableRow.get(CaseDetailedTableViewHeaders.OUTCOME_OF_CASE.toString()),
+              "Outcome of case status is invalid");
+          softly.assertAll();
+        });
+
+    When(
+        "I filter with ([^\"]*) Case ID",
+        (String option) -> {
+          if (option.equals("first")) {
+            webDriverHelpers.fillInWebElement(
+                CASE_DIRECTORY_DETAILED_PAGE_FILTER_INPUT, randomUUID_first_user.toString());
+            webDriverHelpers.clickOnWebElementBySelector(CASE_APPLY_FILTERS_BUTTON);
+            webDriverHelpers.waitForPageLoadingSpinnerToDisappear(40);
+          }
+          if (option.equals("second")) {
+            webDriverHelpers.fillInWebElement(
+                CASE_DIRECTORY_DETAILED_PAGE_FILTER_INPUT, randomUUID_second_user.toString());
+            webDriverHelpers.clickOnWebElementBySelector(CASE_APPLY_FILTERS_BUTTON);
+            webDriverHelpers.waitForPageLoadingSpinnerToDisappear(40);
+          }
+        });
+
+    When(
+        "I check if created user is on filtered list with ([^\"]*) status",
+        (String status) -> {
+          List<Map<String, String>> tableRowsData = getTableRowsData();
+          Map<String, String> detailedCasesTableRow = tableRowsData.get(0);
+          softly.assertEquals(
+              detailedCasesTableRow.get(CaseDetailedTableViewHeaders.OUTCOME_OF_CASE.toString()),
+              status,
+              "Outcome of case status is invalid");
+          softly.assertAll();
+        });
+
+    When(
+        "I filter Persons by created person name",
+        () -> {
+          TimeUnit.SECONDS.sleep(3);
+          webDriverHelpers.waitForPageLoaded();
```

Comment:
remove all waits and add condition to wait until SEARCH_PERSON_BY_FREE_TEXT is present